### PR TITLE
tiflash: add doc for tiflash spill to disk

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

as title

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/13426
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/tiflash-spill-disk.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    ```
+    [DEBUG] [MemoryTracker.cpp:69] ["Peak memory usage (total): 12.80 GiB."] [source=MemoryTracker] [thread_id=110]
+    ```
+
+## Notes
+
+* When the Hash Aggregation operator does not have a `group by` key, it does not support spilling. Even if the Hash Aggregation operator contains a distinct aggregation function, it does not support spilling.
+* Currently, the threshold is calculated for each operator. If a query contains two Hash Aggregation operators, and the threshold is set to 10 GiB, then the two Hash Aggregation operators will only spill data when the memory usage of the operator itself exceeds 10 GiB.
+* Currently, the Hash Aggregation operators and TopN/Sort operators use the merge aggregation and merge sort algorithm during the restore phase. Therefore, these two operators only triggers a single data spill. If the memory required is very large and the memory usage during the restore phase still exceeds the threshold, the spill will not be triggered again.
```
Comment (windtalker) [inline tiflash/tiflash-spill-disk.md:59]:
```suggestion
* Currently, the Hash Aggregation operators and TopN/Sort operators use the merge aggregation and merge sort algorithm during the restore phase. Therefore, these two operators only triggers a single round of spill. If the memory required is very large and the memory usage during the restore phase still exceeds the threshold, the spill will not be triggered again.
```

File: TOC.md [hunk 2]
```diff
@@ -590,6 +590,7 @@
     - [Use MPP Mode](/tiflash/use-tiflash-mpp-mode.md)
     - [Use FastScan](/tiflash/use-fastscan.md)
     - [Supported Push-down Calculations](/tiflash/tiflash-supported-pushdown-calculations.md)
+    - [Spill to Disk](/tiflash/tiflash-spill-disk.md)
     - [TiFlash Query Result Materialization](/tiflash/tiflash-results-materialization.md)
```
Comment (ran-huang) [inline TOC.md:594]:
```suggestion
    - [TiFlash Query Result Materialization](/tiflash/tiflash-results-materialization.md)
    - [Spill to Disk](/tiflash/tiflash-spill-disk.md)
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Pull request merged**